Add inspection-batch-with-output-gathering example

CermakM opened the pull request:
This workflow submits a Job which creates `n` pod replicas and runs them in a sequence.
Additionally, it mounts an artifact repository to store the resulting artifacts.

NOTE: The artifact repository has to be configured via ConfigMap, see
https://github.com/argoproj/argo/blob/master/ARTIFACT_REPO.md
or use the Argo ansible role: https://github.com/CermakM/ansible-role-argo-workflows

successCondition: `n` pods have finished successfully and are present in the cluster

The workflow simulates the scenario when we want to create a batch of inspections
and store the outputs to the artifact store.

Signed-off-by: Marek Cermak <macermak@redhat.com>

Commit:
Add inspection-batch-with-output-gathering example

This workflow submits a Job which creates `n` pod replicas and runs them in a sequence.
Additionally, it mounts an artifact repository to store the resulting artifacts.

NOTE: The artifact repository has to be configured via ConfigMap, see
https://github.com/argoproj/argo/blob/master/ARTIFACT_REPO.md
or use the Argo ansible role: https://github.com/CermakM/ansible-role-argo-workflows

successCondition: `n` pods have finished successfully and are present in the cluster

The workflow simulates the scenario when we want to create a batch of inspections
and store the outputs to the artifact store.

Signed-off-by: Marek Cermak <macermak@redhat.com>

CermakM commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Create examples of Argo Workflows.](https://trello.com/c/rbSVX7o4/730-create-examples-of-argo-workflows)

Commit:
Use exit handler for output collection

Signed-off-by: Marek Cermak <macermak@redhat.com>

modified:   examples/inspection-batch-with-output-gathering.yaml
modified:   examples/templates/template-collect-outputs.yaml

goern commented on examples/inspection-batch-with-output-gathering.yaml:
```diff
@@ -0,0 +1,69 @@
+            spec:
+              containers:
+              - name: fake-inspection
+                image: alpine:latest
```
ubi8!

CermakM replied:
Ye ye :D

Commit:
Fix template for inspection output gathering

- add emptyDirs
- remove trailing spaces
- run as argo service account
- use ubi8 image

Signed-off-by: Marek Cermak <macermak@redhat.com>

modified:   examples/inspection-batch-with-output-gathering.yaml
modified:   examples/templates/template-collect-outputs.yaml

ghost commented:
Build failed.

- [thoth-coala ](https://managesf.thoth-station.ninja/logs/3/3/1bc92e4946ea08fbf0a0f93a185ca7c40bfa8a30/check/thoth-coala/5b5e1ce/) : FAILURE in 37s

goern commented:
@CermakM this needs some love

CermakM commented:
recheck

ghost commented:
Build failed.

- [thoth-coala ](https://managesf.thoth-station.ninja/logs/3/3/1bc92e4946ea08fbf0a0f93a185ca7c40bfa8a30/check/thoth-coala/7780168/) : FAILURE in 48s

Commit:
Fixed missing document start "---"

Signed-off-by: Marek Cermak <macermak@redhat.com>

modified:   examples/inspection-batch-with-output-gathering.yaml
modified:   examples/templates/template-collect-outputs.yaml

ghost commented:
Build succeeded.

- [thoth-coala ](https://managesf.thoth-station.ninja/logs/3/3/27699993868b568eb659a6cf0f13168fcb695c99/check/thoth-coala/bde16c2/) : SUCCESS in 36s

ghost commented:
Build succeeded (gate pipeline).

- [thoth-coala ](https://managesf.thoth-station.ninja/logs/3/3/27699993868b568eb659a6cf0f13168fcb695c99/gate/thoth-coala/63ed6c0/) : SUCCESS in 42s
